pyrandwalk/pyrandwalk_obj.py

```diff
@@ -84,20 +84,30 @@ def prob_sec(self, sequence, initial_dist=None):
             current_state = next_state
         return probability
 
-    def run(self, ntimes=10, show=False):
+    def run(self, ntimes:int=10, show:bool=False, starting_state:int=None):
         """
         Run random walk for ntimes and print out result on each transition.
 
         :param ntimes: numbers of running
         :type ntimes: int
         :param show: flag which is set when showing run is desired
+        :starting_state: the state from which the randon walk should start from
         :return: (generated_states, probability)
         """
-        state = np.random.choice(self.S)
+        if starting_state == None:
+            state = np.random.choice(self.S)
+        else:
+            state = starting_state
+            
         states = [state]
         probabilities = [1 / len(self.S)]
         for i in range(ntimes):
-            state_idx = self.S.index(state)
+            try:
+                state_idx = self.S.index(state)
+            except Exception as e:
+                print("The following error occured:")
+                print(e)
+                print("*HINT*: Make sure <states> is a List or an iterable object with indecies")
```

Comment:
Please follow the best practice for adding an exception. You can take one of our exceptions [here](https://github.com/sadrasabouri/pyrandwalk/blob/master/pyrandwalk/pyrandwalk_error.py) as an example.

Reply:
Thanks for the pointer! I had not gone through the rest of your code to see the exception template classes. Will define a `run()`-specific exception child class and will handle exceptions using that instead!

Reply:
I actually realized that it would be better to take care of a different issue instead of catching exceptions in the `run()` method. I raised an [issue](https://github.com/sadrasabouri/pyrandwalk/issues/16) to address the problem. I will do another PR after this one to attempt fixing the problem I raised. So I have removed this portion from the next [commit](https://github.com/sadrasabouri/pyrandwalk/pull/15/commits/bf69bb560bf0fe182e3073c258842ee5b65cd964).